inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -309,7 +309,8 @@ class ScaleFactorPerLayer<InferenceEngine::EltwiseLayer*> {
             case InferenceEngine::EltwiseLayer::Sub:
             case InferenceEngine::EltwiseLayer::Sum: {
                 // detect which input will be used as biases
-                if (LayerInfo(in0).has32BOutput()) {
+                if (LayerInfo(in0).has32BOutput() ||
+                    (LayerInfo(in0).isNonFunctional() && LayerInfo(InferenceEngine::CNNNetPrevLayer(in0, 0)).has32BOutput())) {
```

Comment:
What happens in cases:
Eltwise -> reshape -> permute -> reshape -> eltwise
this check will return false even though it should be true, you should make an upstream search until you find first functional layer and then checks its output and not only one layer above